document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreTest.java

```diff
@@ -629,6 +630,77 @@ public void testSubDocumentUpdate(String dataStoreName) throws IOException {
     Assertions.assertEquals(expected, OBJECT_MAPPER.writeValueAsString(jsonNode));
   }
 
+  @ParameterizedTest
+  @MethodSource("databaseContextProvider")
+  public void bulkUpdateSubDocForNonExisting(String dataStoreName) throws Exception {
+    Datastore datastore = datastoreMap.get(dataStoreName);
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+    Map<Key, Map<String, Document>> toUpdate = new HashMap<>();
+    Key key1 = new SingleValueKey("tenant-1", "testKey1");
+    Key key2 = new SingleValueKey("tenant-2", "testKey2");
+    Map<String, Document> subDoc1 = new HashMap<>(), subDoc2 = new HashMap<>();
+    subDoc1.put("subDocPath1", Utils.createDocument("timestamp", "100"));
+    subDoc2.put("subDocPath2", Utils.createDocument("timestamp", "100"));
+    toUpdate.put(key1, subDoc1);
+    toUpdate.put(key2, subDoc2);
+    int result = collection.bulkUpdateSubDoc(toUpdate);
+    assertEquals(0, result);
+  }
+
+  @ParameterizedTest
+  @MethodSource("databaseContextProvider")
+  public void bulkUpdateSubDocForEmptyMap(String dataStoreName) throws Exception {
+    Datastore datastore = datastoreMap.get(dataStoreName);
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+    Map<Key, Map<String, Document>> toUpdate = new HashMap<>();
+    int result = collection.bulkUpdateSubDoc(toUpdate);
+    assertEquals(0, result);
+  }
+
+  @ParameterizedTest
+  @MethodSource("databaseContextProvider")
+  public void bulkUpdateSubDocOnlyForExisting(String dataStoreName) throws Exception {
+    Datastore datastore = datastoreMap.get(dataStoreName);
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+    Key key1 = new SingleValueKey("tenant-1", "testKey1");
+    Key key2 = new SingleValueKey("tenant-2", "testKey2");
+    Key key3 = new SingleValueKey("tenant-3", "testKey3");
+    collection.upsert(key1, Utils.createDocument("foo1", "bar1"));
+    collection.upsert(key3, Utils.createDocument("foo3", "bar3"));
+
+    Map<Key, Map<String, Document>> toUpdate = new HashMap<>();
+    Map<String, Document> subDoc1 = new HashMap<>(),
+        subDoc2 = new HashMap<>(),
+        subDoc3 = new HashMap<>();
+    subDoc1.put("subDocPath1", Utils.createDocument("nested1", "100"));
+    subDoc2.put("subDocPath2", Utils.createDocument("nested2", "100"));
+    subDoc3.put("subDocPath3", Utils.createDocument("nested3", "100"));
```

Comment:
Can you also add a test, where it updates an existing sub doc path with a new document?